tls-tutorial/doc/use_your_own_certificate.md

```diff
@@ -0,0 +1,685 @@
+
+We keep things simple here though, since this is just a tutorial meant to convey the general idea of server certificate validation.
+
+## Setting up MySQL
+
+1. Install MySQL client and server
+   ```shell
+   sudo apt install mysql-client mysql-server
+   ```
+2. Log into MySQL as MySQL root user (on Ubuntu now requires `sudo`)
+   ```shell
+   sudo mysql -u root
+   ```
+3. Create a database `shapes` and a user `tutorial_user` for our import experiment
+   ```sql
+   create database shapes;
+   create user 'tutorial_user' identified by 'tutorial';
+   grant all on shapes.* to tutorial_user;
+   exit
+   ```
+4. Login as a `tutorial_user`
+   ```sql
+   mysql -u tutorial_user -p
+   ```
+5. Populate the database
+   ```sql
+   create table shapes.shapes (name VARCHAR(40), corners int);
+   insert into shapes.shapes values ('point', 1), ('line', 2), ('triangle', 3), ('rectangle', 4);
+   exit
+   ```
+
+As you can tell we won't win any price for most original database usage here, but that is not the point of the tutorial.
+
+### Installing Certificates in MySQL
+
+By default, the MySQL server on Ubuntu comes with a self-signed certificate. Our next job will be to use the certificates we previously created.
+
+1. Install the CA certificate centrally:
+   ```shell
+   sudo cp ca.crt /usr/local/share/ca-certificates/exasol_tutorial_ca.crt
+   sudo update-ca-certificates
+   ```
+2. Verify that it was installed:
+   ```shell
+   ls -l /etc/ssl/certs/exasol*
+   ```
+   Must output something like:
+   ```
+   lrwxrwxrwx 1 root root 55 Jan 12 11:43 /etc/ssl/certs/exasol_tutorial_ca.pem -> /usr/local/share/ca-certificates/exasol_tutorial_ca.crt
+   ```
+   Note how the symbolic link changes the file suffice from `.crt` to `.pem`
+3. Install the server certificate and key
+   ```shell
+   sudo mkdir -p /etc/mysql/certs
+   sudo cp server.crt /etc/mysql/certs/
+   sudo cp server.key /etc/mysql/certs/
+   sudo chown mysql:mysql /etc/mysql/certs/server.key
+   ```
+
+Note that we need `sudo` privileges here, since we need to install the certificates in a central locations that belong to the `root` user.
+
+Additionally, we give the private key to the MySQL user. We don't want anyone else to be able to read it, so weakening permissions on that file is not an option.
+
+You can check ownership and permissions like this:
+
+```shell
+ls -l /etc/mysql/certs/
+```
+
+This should list two files:
+
+```
+-rw-r--r-- 1 root  root  1919 Jan 12 11:56 server.crt
+-rw------- 1 mysql mysql 3326 Jan 12 11:56 server.key
+```
+
+Now edit the `[mysqld]` section of the MySQL configuration file and *add* the following lines as `root` user:
+
+```
+[mysqld]
+ssl
+ssl-ca=/etc/ssl/certs/exasol_tutorial_ca.pem
+ssl-cert=/etc/mysql/certs/server.crt
+ssl-key=/etc/mysql/certs/server_key.pem
+```
+
+Also, we need to enable Exasol later to access the MySQL server. Since that will not be access from `localhost` you need to accept all interfaces.
+
+```
+bind-address = 0.0.0.0
```

Comment:
```suggestion
Now edit the `[mysqld]` section of the MySQL configuration file /etc/mysql/mysql.conf.d/mysqld.cnf as `root` user:

Add
```
[mysqld]
ssl
ssl-ca=/etc/ssl/certs/exasol_tutorial_ca.pem
ssl-cert=/etc/mysql/certs/server.crt
ssl-key=/etc/mysql/certs/server_key.pem
```

Also, we need to enable Exasol later to access the MySQL server. Since that will not be access from `localhost` you need to accept all interfaces.

Change
```
bind-address = 0.0.0.0
```